docs/content/sips/xxx-spin-state-wasi.md

```diff
@@ -0,0 +1,131 @@
+title = "SIP xxx - Logs and Metadata WASI filesystem"
+template = "main"
+date = "2022-06-07T22:49:37"
+
+---
+
+Summary: Expose the logs and metadata to components as a filesystem
+
+Owner: me@mitchellhynes.com
+
+Created: June 7th, 2022
+
+Updated: June 8th, 2022
+
+## Background
+
+Apart from reading the file system of the host machine, there's no way to read
+logs or manifest of a running Spin app. This means users are limited in what
+they know about a Spin app while it's running, unless they have access to the
+environment it is on.
+
+[Laundromat](https://github.com/ecumene/laundromat) is a tool for reading logs
+and organizing them in a quicker format for developer velocity, and it gets
+around this by serving the logs of a Spin app as a REST API. This is limited
+because
+
+- it must have access to the file system which users may not have access to
+- being a server itself it must be forwarded out of whatever network users are
+  running it in, along with Spin itself
+- the file system is itself limited as new logs won't notify the user.
+  system file events could be used but they're notoriously platform dependent.
+
+If Spin exposed a "monitoring filesystem" that was accessible by WASI FS we
+could open the door to many hypervisor-like tools.
+
+### The `~/.spin/` directory
+
+The `~/.spin/` directory is where Spin keeps logs and other ephemeral data for
+apps.
+
+```
+tree ~/.spin/
+└── appname
+    ├── key-value
+    └── logs
+        ├── componentname_stderr.txt
+        └── componentname_stdout.txt
+```
+
+## Proposal
+
+WASI provides a system interface where components can access files as if they
+were system files in their program, except those files don't have to be on disk
+and don't even have to be files at all.
+
+### `~/.spin/` 2: Escape 2 Madigascar
+
+Exposing `~/.spin/` to components doesn't work because all files and folders
+remain in the same state they were mounted at. Any new logs won't be included
+when read again.
+
+If instead of mounting a `~/.spin/` directory on the host machine, we allowed
+users to mount a directory internal to Spin in-memory this would be updated
+on subsequent reads while having the access control of a mounted directory.
+
+#### Mount: `/monitor/`
```

Comment:
I think we should namespace this _within_ the WASI FS to avoid conflicting with a component's own mounts. I think `/.spin/monitor/` would be sensible and give us a place to put future host-provided files; I have thought about exposing config variables in a similar way and those could go in `/.spin/config/`.